packages/spectacle/src/components/typography.test.tsx

```diff
@@ -91,3 +92,55 @@ describe('<CodeSpan />', () => {
     expect(container.querySelector('code')?.innerHTML).toBe('Code!');
   });
 });
+
+describe('<FitText />', () => {
+  beforeEach(() => {
+    // Default mock implementation
+    jest.mock('use-resize-observer', () => {
+      return { width: 500, height: 100 };
+    });
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should render text content correctly', () => {
+    const { getByText } = mountWithTheme(<FitText>Spectacle!</FitText>);
+    expect(getByText('Spectacle!')).toBeInTheDocument();
+  });
+
+  it('should apply color and typography props correctly', () => {
+    const { getByText } = mountWithTheme(
+      <FitText color="secondary" fontSize="h1">
+        Spectacle!
+      </FitText>
+    );
+    const textElement = getByText('Spectacle!');
+    expect(textElement).toHaveStyle({ color: defaultTheme.colors.secondary });
+    expect(textElement).toHaveStyle({ fontSize: 'h1' });
+  });
+
+  it('should scale text when container size changes', () => {
+    // Simulate a container that's smaller than the text
+    jest.mock('use-resize-observer', () => {
+      return { width: 100, height: 100 };
+    });
+
+    const { container } = mountWithTheme(
+      <FitText>Long text that needs scaling</FitText>
+    );
+    const scaledText = container.querySelector('div[scale]');
+    expect(scaledText).toHaveStyle({ transform: 'scale(1)' }); // TODO: Not sure if this is a great test / expected.
```

Comment:
I think this is tricky to test because it's real hard to consistently simulate metrics that involve DOM and fonts. I think I would just see that the function that applies and calculates a new scale is called when the container is resized.

Reply:
I dug into this more and as we already have to fully polyfill `use-resize-observer` it's really hard to get a test that isn't just testing the mocks itself.

I ended up removing the test for not really testing anything...